emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -48,103 +96,239 @@ public class VmsMojo extends DiffMojo {
+     * @throws MojoExecutionException if the Surefire execution fails.
+     */
+    private void invokeSurefire() throws MojoExecutionException {
+        InvocationRequest request = new DefaultInvocationRequest();
+        request.setGoals(Collections.singletonList("surefire:test"));
+        InvocationOutputHandler outputHandler = new SurefireOutputHandler(new ViolationFilterer());
+        request.setOutputHandler(outputHandler);
+        request.setErrorHandler(outputHandler);
+
+        try {
+            Invoker invoker = new DefaultInvoker();
+            InvocationResult result = invoker.execute(request);
+
+            if (result.getExitCode() != 0) {
+                throw new MojoExecutionException("Surefire reported exit code " + result.getExitCode());
+            }
+        } catch (MavenInvocationException ex) {
+            throw new MojoExecutionException("Failed to execute Surefire", ex);
+        }
     }
 
     /**
      * Fetches the two most recent commits of the repository and finds the differences between them.
+     * If the user has specified a lastSha, then that sha will be used as the "old" version of code. It should ideally
+     * correspond to
      *
      * @return List of differences between two most recent commits of the repository
      * @throws MojoExecutionException if error is encountered at runtime
      */
     private List<DiffEntry> getCommitDiffs() throws MojoExecutionException {
-        Git git;
-        Iterable<RevCommit> commits;
         ObjectReader objectReader;
         List<DiffEntry> diffs;
+        List<AbstractTreeIterator> trees = new ArrayList<>();
 
-        // Sets up repository and fetches commits
-        try {
-            git = Git.open(basedir.toPath().resolve(".git").toFile());
-            commits = git.log().setMaxCount(1).call();
+        try (Git git = Git.open(gitDir.toFile())) {
             objectReader = git.getRepository().newObjectReader();
-        } catch (GitAPIException | IOException exception) {
-            throw new MojoExecutionException("Failed to fetch two previous commits from repository");
-        }
 
-        // Creates trees to parse through to analyze for differences
-        List<AbstractTreeIterator> trees = new ArrayList<>();
-        trees.add(new FileTreeIterator(git.getRepository()));
-        try {
-            for (RevCommit commit : commits) {
-                trees.add(new CanonicalTreeParser(null, objectReader, commit.getTree().getId()));
+            // Set up diffFormatter
+            diffFormatter.setRepository(git.getRepository());
+            diffFormatter.setContext(0);
+            diffFormatter.setDetectRenames(true);
+
+            // Get more recent version of code (either working tree or most recent commit)
+            if (newSha == null || newSha.isEmpty()) {
+                trees.add(new FileTreeIterator(git.getRepository()));
+            } else {
+                ObjectId shaId = git.getRepository().resolve(lastSha);
```

Comment:
Why is this not `newSha`?

Reply:
Because I made a mistake :) 
Fixed in latest commit